### PR TITLE
Update CHANGES and NEWS for 1.1.1h release

### PR DESCRIPTION
**Pull request by mspncp**
Citing pull request #11346:

> Since the grand markdown revamping in #10545, the CHANGES and NEWS files contain notes for _all_ releases, so they need to be updated, too.
> 
> As long as there is only one stable branch, I can take care of it every time a release has happened.
> 
> _Note: For 3.0 and beyond, the ideal solution would be to update the CHANGES and NEWS files on master only, and backport the relevant sections to the respecitve stable branches using a script when creating the release. But that's a task for a separate pull request in the future..._
> 
> cc @mattcaswell

Since then, the markdown versions on the master branch haven't been updated, which was reported by Michael Wojcik on [openssl-users](https://mta.openssl.org/pipermail/openssl-users/2020-September/012984.html). This is fixed by the first commit.

> Author: Dr. Matthias St. Pierre <matthias.st.pierre@ncp-e.com>
> Date:   Thu Sep 24 07:58:52 2020 +0200
> 
>     Update CHANGES and NEWS for 1.1.1h release

While working on it, I adjusted the CVE link style, see second commit.

> Author: Dr. Matthias St. Pierre <matthias.st.pierre@ncp-e.com>
> Date:   Thu Sep 24 08:11:00 2020 +0200
> 
>     Change CVE link style in CHANGES and NEWS
>     
>     Replace [collapsed reference links][] for the CVEs by
>     [shortcut reference links], in order to to improve the
>     readability of the raw markdown text.
>     
>     Consistently add parentheses around the CVE links at the
>     end of the CVE descriptions. (The NEWS file already had
>     the parentheses, in the CHANGES file they where missing.)
>     
>     [collapsed reference links]:
>       https://github.github.com/gfm/#collapsed-reference-link
>     
>     [shortcut reference links]:
>       https://github.github.com/gfm/#shortcut-reference-link

### CODE REVIEW

File: CHANGES.md
Comment (mspncp) [inline CHANGES.md:1202]: Ups, I overlooked to replace some of the brackets around the names. Will do it later, no time left.

Comment (mspncp) [reply CHANGES.md]: Done

**Comment by mspncp** [comment]
@mattcaswell when and how does the [changelog](https://www.openssl.org/news/changelog.html) on the OpenSSL website get updated?

**Comment by mattcaswell** [comment]
> @mattcaswell when and how does the changelog on the OpenSSL website get updated?

I think it gets updated automatically whenever we do a release. Or we can make it happen manually.

**Comment by gregbown** [comment]
Unsure why this is but when downloading the 1.1.1-h release from the [openssl website](https://www.openssl.org/source/openssl-1.1.1h.tar.gz) and building from the download the version that is built is not 1.1.1-h

The version built from 1.1.1-h download is:
```
OpenSSL 1.1.1b  26 Feb 2019
```

**Comment by mspncp** [comment]
@gregbown there are in general two ways how you can end up with the wrong openssl version: you either executed the wrong openssl executable, or the excutable loaded the wrong openssl libraries.

Please check whether you executed the correct binary

```
which openssl

```
and whether your binary loaded the correct shared libraries:

```
ldd $(which openssl)
```

**Note:** a version mismatch between application and library would be detected automatically by `openssl version` and displayed as follows:

```
OpenSSL 1.1.1h  22 Sep 2020 (Library: OpenSSL 1.1.1d  10 Sep 2019)
```

**Comment by mspncp** [comment]
BTW: this pull request is still waiting for some review @openssl/committers.

**Review by beldmit — APPROVED** [review]
LGTM

**Comment by gregbown** [comment]
@mspncp Regarding building 1.1.1-h Thank you for taking the time to educate me regarding inspecting the binary that was loaded. Ironically my system was loading the version that came with the latest Perl installation that I installed to build openssl.  Than you again

**Comment by mspncp** [comment]
Merged to master, thank you for the review!

* d8dc853825 Change CVE link style in CHANGES and NEWS
* 6ffc312776 Update CHANGES and NEWS for 1.1.1h release

**Pull request closed without merging**
